<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainForm</class>
 <widget class="QMainWindow" name="MainForm">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>934</width>
    <height>729</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Super Mega ML</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QTabWidget" name="tabWidget">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>0</y>
      <width>941</width>
      <height>731</height>
     </rect>
    </property>
    <property name="currentIndex">
     <number>0</number>
    </property>
    <widget class="QWidget" name="tabSetup">
     <attribute name="title">
      <string>Setup</string>
     </attribute>
     <widget class="QGroupBox" name="groupData">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>0</y>
        <width>461</width>
        <height>691</height>
       </rect>
      </property>
      <property name="title">
       <string>Data Parameters</string>
      </property>
      <widget class="QGroupBox" name="groupBox">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>40</y>
         <width>441</width>
         <height>471</height>
        </rect>
       </property>
       <property name="title">
        <string>Internal Datasets</string>
       </property>
       <widget class="QComboBox" name="comboDataset">
        <property name="geometry">
         <rect>
          <x>145</x>
          <y>40</y>
          <width>281</width>
          <height>26</height>
         </rect>
        </property>
        <item>
         <property name="text">
          <string>Linear 2D with noise</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>&quot;Fish&quot; 2D</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Linear 3D with noise</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>&quot;Fish&quot; 3D</string>
         </property>
        </item>
       </widget>
       <widget class="QLabel" name="label_15">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>40</y>
          <width>131</width>
          <height>31</height>
         </rect>
        </property>
        <property name="text">
         <string>Dataset</string>
        </property>
       </widget>
       <widget class="QWidget" name="widgetDataset" native="true">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>80</y>
          <width>421</width>
          <height>381</height>
         </rect>
        </property>
       </widget>
      </widget>
     </widget>
     <widget class="QGroupBox" name="groupModel">
      <property name="geometry">
       <rect>
        <x>470</x>
        <y>0</y>
        <width>461</width>
        <height>691</height>
       </rect>
      </property>
      <property name="title">
       <string>Model Parameters</string>
      </property>
      <widget class="QComboBox" name="comboModel">
       <property name="geometry">
        <rect>
         <x>165</x>
         <y>30</y>
         <width>231</width>
         <height>26</height>
        </rect>
       </property>
       <item>
        <property name="text">
         <string>Linear Regression</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>Random</string>
        </property>
       </item>
      </widget>
      <widget class="QLabel" name="label_8">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>30</y>
         <width>141</width>
         <height>31</height>
        </rect>
       </property>
       <property name="text">
        <string>Model</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_9">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>70</y>
         <width>140</width>
         <height>31</height>
        </rect>
       </property>
       <property name="text">
        <string>Optimizer</string>
       </property>
      </widget>
      <widget class="QComboBox" name="comboOptimizer">
       <property name="geometry">
        <rect>
         <x>165</x>
         <y>70</y>
         <width>231</width>
         <height>26</height>
        </rect>
       </property>
       <item>
        <property name="text">
         <string>Gradient</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>SGD</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>MomentGrad</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>NesterovGrad</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>MyGrad</string>
        </property>
       </item>
      </widget>
      <widget class="QLabel" name="label_10">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>110</y>
         <width>141</width>
         <height>31</height>
        </rect>
       </property>
       <property name="text">
        <string>Loss Function</string>
       </property>
      </widget>
      <widget class="QComboBox" name="comboLoss">
       <property name="geometry">
        <rect>
         <x>165</x>
         <y>110</y>
         <width>231</width>
         <height>26</height>
        </rect>
       </property>
       <item>
        <property name="text">
         <string>Linear MSE</string>
        </property>
       </item>
      </widget>
      <widget class="QPlainTextEdit" name="plainOptimizer">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>280</y>
         <width>441</width>
         <height>64</height>
        </rect>
       </property>
       <property name="plainText">
        <string/>
       </property>
      </widget>
      <widget class="QLabel" name="label_11">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>250</y>
         <width>441</width>
         <height>31</height>
        </rect>
       </property>
       <property name="text">
        <string>Optimizer Options</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignBottom|Qt::AlignHCenter</set>
       </property>
      </widget>
      <widget class="QLabel" name="label_12">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>350</y>
         <width>441</width>
         <height>31</height>
        </rect>
       </property>
       <property name="text">
        <string>Loss Options</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignBottom|Qt::AlignHCenter</set>
       </property>
      </widget>
      <widget class="QPlainTextEdit" name="plainLoss">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>380</y>
         <width>441</width>
         <height>64</height>
        </rect>
       </property>
       <property name="readOnly">
        <bool>false</bool>
       </property>
      </widget>
      <widget class="QLabel" name="label_13">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>150</y>
         <width>441</width>
         <height>31</height>
        </rect>
       </property>
       <property name="text">
        <string>Model Options</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignBottom|Qt::AlignHCenter</set>
       </property>
      </widget>
      <widget class="QPlainTextEdit" name="plainModel">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>180</y>
         <width>441</width>
         <height>64</height>
        </rect>
       </property>
       <property name="readOnly">
        <bool>false</bool>
       </property>
      </widget>
     </widget>
    </widget>
    <widget class="QWidget" name="tabTrain">
     <attribute name="title">
      <string>Training</string>
     </attribute>
     <widget class="QGroupBox" name="groupTrain">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>0</y>
        <width>271</width>
        <height>691</height>
       </rect>
      </property>
      <property name="title">
       <string>Training</string>
      </property>
      <widget class="QLabel" name="label">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>40</y>
         <width>111</width>
         <height>31</height>
        </rect>
       </property>
       <property name="text">
        <string>Epoch Limit</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
       </property>
      </widget>
      <widget class="QSpinBox" name="spinEpochLimit">
       <property name="geometry">
        <rect>
         <x>130</x>
         <y>40</y>
         <width>131</width>
         <height>30</height>
        </rect>
       </property>
       <property name="maximum">
        <number>200000</number>
       </property>
       <property name="singleStep">
        <number>100</number>
       </property>
       <property name="value">
        <number>200</number>
       </property>
      </widget>
      <widget class="QLabel" name="label_2">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>80</y>
         <width>131</width>
         <height>31</height>
        </rect>
       </property>
       <property name="text">
        <string>Learning Rate</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
       </property>
      </widget>
      <widget class="QDoubleSpinBox" name="spinLearningRate">
       <property name="geometry">
        <rect>
         <x>140</x>
         <y>80</y>
         <width>121</width>
         <height>30</height>
        </rect>
       </property>
       <property name="decimals">
        <number>5</number>
       </property>
       <property name="minimum">
        <double>0.000010000000000</double>
       </property>
       <property name="maximum">
        <double>2.000000000000000</double>
       </property>
       <property name="singleStep">
        <double>0.001000000000000</double>
       </property>
      </widget>
      <widget class="QLabel" name="label_3">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>120</y>
         <width>131</width>
         <height>31</height>
        </rect>
       </property>
       <property name="text">
        <string>Satis. Loss</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
       </property>
      </widget>
      <widget class="QDoubleSpinBox" name="spinSatisLoss">
       <property name="geometry">
        <rect>
         <x>140</x>
         <y>120</y>
         <width>121</width>
         <height>30</height>
        </rect>
       </property>
       <property name="decimals">
        <number>5</number>
       </property>
       <property name="minimum">
        <double>0.000010000000000</double>
       </property>
       <property name="maximum">
        <double>2.000000000000000</double>
       </property>
       <property name="singleStep">
        <double>0.001000000000000</double>
       </property>
      </widget>
      <widget class="QLabel" name="label_4">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>160</y>
         <width>131</width>
         <height>31</height>
        </rect>
       </property>
       <property name="text">
        <string>Minimal DLoss</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
       </property>
      </widget>
      <widget class="QDoubleSpinBox" name="spinMinDLoss">
       <property name="geometry">
        <rect>
         <x>140</x>
         <y>160</y>
         <width>121</width>
         <height>30</height>
        </rect>
       </property>
       <property name="decimals">
        <number>5</number>
       </property>
       <property name="minimum">
        <double>0.000010000000000</double>
       </property>
       <property name="maximum">
        <double>2.000000000000000</double>
       </property>
       <property name="singleStep">
        <double>0.001000000000000</double>
       </property>
       <property name="value">
        <double>0.000010000000000</double>
       </property>
      </widget>
      <widget class="QCheckBox" name="checkHotStart">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>370</y>
         <width>131</width>
         <height>31</height>
        </rect>
       </property>
       <property name="text">
        <string>Hot Start</string>
       </property>
       <property name="checkable">
        <bool>false</bool>
       </property>
      </widget>
      <widget class="QPushButton" name="buttonFit">
       <property name="geometry">
        <rect>
         <x>130</x>
         <y>370</y>
         <width>131</width>
         <height>30</height>
        </rect>
       </property>
       <property name="text">
        <string>Fit</string>
       </property>
      </widget>
      <widget class="QPlainTextEdit" name="plainFit">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>230</y>
         <width>251</width>
         <height>131</height>
        </rect>
       </property>
      </widget>
      <widget class="QLabel" name="label_14">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>200</y>
         <width>251</width>
         <height>31</height>
        </rect>
       </property>
       <property name="text">
        <string>Fit Parameters</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignBottom|Qt::AlignHCenter</set>
       </property>
      </widget>
      <widget class="QWidget" name="widgetLoss" native="true">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>460</y>
         <width>251</width>
         <height>221</height>
        </rect>
       </property>
      </widget>
      <widget class="QSpinBox" name="spinLossHistory">
       <property name="geometry">
        <rect>
         <x>130</x>
         <y>410</y>
         <width>131</width>
         <height>30</height>
        </rect>
       </property>
       <property name="maximum">
        <number>200000</number>
       </property>
       <property name="singleStep">
        <number>10</number>
       </property>
      </widget>
      <widget class="QPushButton" name="buttonLossHist">
       <property name="geometry">
        <rect>
         <x>14</x>
         <y>410</y>
         <width>111</width>
         <height>30</height>
        </rect>
       </property>
       <property name="text">
        <string>View last:</string>
       </property>
      </widget>
     </widget>
     <widget class="QGroupBox" name="groupResult">
      <property name="geometry">
       <rect>
        <x>280</x>
        <y>0</y>
        <width>651</width>
        <height>691</height>
       </rect>
      </property>
      <property name="title">
       <string>Results</string>
      </property>
      <widget class="QWidget" name="widgetResults" native="true">
       <property name="geometry">
        <rect>
         <x>9</x>
         <y>29</y>
         <width>631</width>
         <height>651</height>
        </rect>
       </property>
      </widget>
     </widget>
    </widget>
   </widget>
  </widget>
 </widget>
 <resources/>
 <connections/>
</ui>
